PassRetrievalTool.py

```diff
@@ -0,0 +1,133 @@
+import json
+import locale
+import subprocess
+import re
+import os
+import sys
+
+
+class PassRetrievalTool:
+    # Static Variables
+    COMMAND_LINUX = "sudo grep -r '^psk=' /etc/NetworkManager/system-connections/"
+    COMMAND_OSX = "defaults read /Library/Preferences/SystemConfiguration/com.apple.airport.preferences |grep SSIDString"
+    COMMAND_WINDOWS_GENERIC = "netsh wlan show profile"
+    RE_LINUX = '/etc/NetworkManager/system-connections/(.*)'
+    RE_OSX = 'SSIDString = (.*);'
+    PASS_OSX = 'security find-generic-password -wa '
+    LANGUAGE_STRINGS = {
+        "en_GB" : "Key Content",
+        "en_US" : "Key Content",
+        "es_ES" : "Contenido de la clave"
+    }
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def get_pass_wind_individual(Name):
+        output = subprocess.check_output(PassRetrievalTool.COMMAND_WINDOWS_GENERIC + " name=" + Name + " key=clear", shell=True)
+        output = re.findall(PassRetrievalTool.getLanguageString() + '(.*)\n', output)[0].strip().split(':')[1].strip()
+        return output
+
+    @staticmethod
+    def get_all_wifi_password_dictionary():
+        return PassRetrievalTool.get_specific_wifi_password_dictionary(None)
+    @staticmethod
+    def get_specific_wifi_password_dictionary(filter_ssid):
+        wifi_password_dictionary = dict()
+        if os.name == 'posix':
+            try:
+                output = subprocess.check_output(PassRetrievalTool.COMMAND_LINUX, shell=True).split('\n')
+                for pair in output:
+                    try:
+                        pair = re.findall(PassRetrievalTool.RE_LINUX, pair)[0].split(':')
+                        name = pair[0]
+                        password = pair[1].split('=')[1]
+
+                        if filter_ssid and name != filter_ssid:
+                            continue # Ignore if filter is set and doesn't match
+
+                        wifi_password_dictionary[name] = password
+                    except:
+                        pass
+            except:
+                output = subprocess.check_output(PassRetrievalTool.COMMAND_OSX, shell=True).split('\n')
+                for pair in output:
+                    try:
+                        name = re.findall(PassRetrievalTool.RE_OSX, pair)[0]
+                        password = subprocess.check_output(PassRetrievalTool.PASS_OSX + name, shell=True)
+
+                        if filter_ssid and name != filter_ssid:
+                            continue # Ignore if filter is set and doesn't match
```

Comment:
Instead of ignoring, print out something that tells the user that the ssid he's requesting the password for is not in the system.